### PR TITLE
CORE:

### PR DESCRIPTION
**Pull request by anonthing**
More fixes for TimeRouter

### CODE REVIEW

File: core/src/main/java/net/opentsdb/query/processor/merge/Merger.java [hunk 1]
```diff
@@ -146,7 +146,7 @@ public Merger(final QueryNodeFactory factory,
                 .build();
       }
     }
-
+    LOG.info("Expect to have results: " + results);
```
Comment (manolama) [inline core/src/main/java/net/opentsdb/query/processor/merge/Merger.java:149]: debug

File: core/src/main/java/net/opentsdb/query/processor/merge/Merger.java [hunk 2]
```diff
@@ -180,6 +180,7 @@ public void onNext(final QueryResult next) {
     }
 
     int countdown = 0;
+    LOG.info("Got response from datasource: " +next.dataSource()  + " and source: " + next.source());
```
Comment (manolama) [inline core/src/main/java/net/opentsdb/query/processor/merge/Merger.java:183]: debug

File: core/src/main/java/net/opentsdb/query/router/RoutingUtils.java [hunk 3]
```diff
@@ -230,20 +230,27 @@ public static void rebuildGraph(
...
-                     "_" + builder.id();
-             pdBuilder.setResultIds(Lists.newArrayList(
-                     new DefaultQueryResultId(
-                            pushdownId,
-                            builder.dataSource())));
-             pdBuilder.setId(pushdownId);
-           } else {
-             pdBuilder.setResultIds(Lists.newArrayList(
+
+//           if (x + 1 == builder.pushDowns().size()) {
```
Comment (manolama) [inline core/src/main/java/net/opentsdb/query/router/RoutingUtils.java:234]: leave a note about why we commented it out.

**Review by manolama — APPROVED** [review]
couple of nits but good

File: core/src/main/java/net/opentsdb/query/processor/merge/Merger.java (hunk 1 above)
Comment (anonthing) [reply core/src/main/java/net/opentsdb/query/processor/merge/Merger.java:149]: want to have it under info for now. doesn't generate too much and will be good to have to visually test the split code

File: core/src/main/java/net/opentsdb/query/processor/merge/Merger.java (hunk 2 above)
Comment (anonthing) [reply core/src/main/java/net/opentsdb/query/processor/merge/Merger.java:183]: same here.

File: core/src/main/java/net/opentsdb/query/router/RoutingUtils.java (hunk 3 above)
Comment (anonthing) [reply core/src/main/java/net/opentsdb/query/router/RoutingUtils.java:234]: oh. I might actually put this back.

File: core/src/main/java/net/opentsdb/query/processor/merge/Merger.java (hunk 1 above)
Comment (manolama) [reply core/src/main/java/net/opentsdb/query/processor/merge/Merger.java:149]: Double check the log configs, I think it's set to debug for prod too :)

**Review by gskchaitanya — APPROVED** [review]

**Comment by manolama** [comment]
@gskchaitanya @anonthing Please do squash commits too when there are a few tweaks like this. Thanks!

**Pull request merged**